src/compiler/types.ts

```diff
@@ -1046,7 +1046,7 @@ namespace ts {
     // @kind(SyntaxKind.JsxOpeningElement)
     export interface JsxOpeningElement extends Expression {
         _openingElementBrand?: any;
-        tagName: EntityName;
+        tagName: LeftHandSideExpression;

```

Comment:
Yea unfortunately with our current structure, we don't have type to describe ThisKeyword. I will rename it be 
`type JsxTagNameExpression = PrimaryExpression | PropertyAccessExpression;`